apisix/plugins/error-log-logger.lua

```diff
@@ -0,0 +1,204 @@
+        max_retry_count = {type = "integer", minimum = 0, default = 0},
+        retry_delay = {type = "integer", minimum = 0, default = 1},
+        buffer_duration = {type = "integer", minimum = 1, default = 60},
+        inactive_timeout = {type = "integer", minimum = 1, default = 3},
+    },
+    required = {"host", "port"}
+}
+
+
+local log_level = {
+    STDERR =    ngx.STDERR,
+    EMERG  =    ngx.EMERG,
+    ALERT  =    ngx.ALERT,
+    CRIT   =    ngx.CRIT,
+    ERR    =    ngx.ERR,
+    ERROR  =    ngx.ERR,
+    WARN   =    ngx.WARN,
+    NOTICE =     ngx.NOTICE,
+    INFO   =    ngx.INFO,
+    DEBUG  =    ngx.DEBUG
+}
+
+
+local config = {}
+local buffers = {}
+
+
+local _M = {
+    version = 0.1,
+    priority = 1091,
+    name = plugin_name,
+    metadata_schema = metadata_schema,
+}
+
+
+local function send_to_server(data)
+    local sock, soc_err = tcp()
+
+    if not sock then
+        return false, "failed to init the socket " .. soc_err
+    end
+
+    sock:settimeout(config.timeout * 1000)
+
+    local ok, err = sock:connect(config.host, config.port)
+    if not ok then
+        return false, "failed to connect the TCP server: host[" .. config.host
+                         .. "] port[" .. tostring(config.port) .. "] err: " .. err
+    end
+
+    if config.tls then
+        ok, err = sock:sslhandshake(true, config.tls_options, false)
+        if not ok then
+            return false, "failed to to perform TLS handshake to TCP server: host["
+                          .. config.host .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        end
+    end
+
+    local bytes, err = sock:send(data)
+    if not bytes then
+        sock:close()
+        return false, "failed to send data to TCP server: host[" .. config.host
+                        .. "] port[" .. tostring(config.port) .. "] err: " .. err
+    end
+
+    sock:setkeepalive(config.keepalive * 1000)
+    return true
+end
+
+
+local function update_filter(value)
+    local level = log_level[string.upper(value.level)]
+    local status, err = errlog.set_filter_level(level)
+    if not status then
+        return nil, "failed to set filter level by ngx.errlog, the error is :" .. err
+    else
+        core.log.debug("set the filter_level to ", config.level)
+    end
+
+    return value
+end
+
+
+local function process()
+    local metadata = plugin.plugin_metadata(plugin_name)
+    if not (metadata and metadata.value and metadata.modifiedIndex) then
+        core.log.info("please set the correct plugin_metadata for ", plugin_name)
+        return
+    else
+        local err
+        config, err = lrucache(plugin_name, metadata.modifiedIndex, update_filter,
+                                 metadata.value)
+        if not config then
+            core.log.warn("set log filter failed for ", err)
+            return
+        end
+
+    end
+
+    local id = ngx.worker.id()
+    local entries = {}
+    local logs = errlog.get_logs(10)
```

Comment:
Better to get 3's multiple? Like `get_logs(9)`. And we should reuse `logs` as the second argument of `get_logs`.

Reply:
LGTM, I'll update it.